src/components/RestroPage/RestroDetails/RestroDetails.js

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from "react";
+import "./RestroDetails.css"
+ class RestroDetails extends Component {
+    constructor(){
+        super();
+        this.state={
+            cart: [],
+            dishList:[
+                {
+                    id: 1,
+                    details: {
+                      Quantity:0,
+                      Name: "Veg Pizza",
+                      Image:"https://media.istockphoto.com/photos/cheesy-pepperoni-pizza-picture-id938742222?k=20&m=938742222&s=612x612&w=0&h=X5AlEERlt4h86X7U7vlGz3bDaDDGQl4C3MuU99u2ZwQ=",
+                      Description:
+                        "Italian Dish",
+                      Price: 200
+                    }
+                  },
+                  {
+                    
+                    id: 2,
+                    details: {
+                      Quantity:0,
+                      Name: "Noodle",
+                      Image:"https://images.unsplash.com/photo-1600490036275-35f5f1656861?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8bm9vZGxlfGVufDB8fDB8fA%3D%3D&w=1000&q=80",
+                      Description:
+                        "Chinese",
+                      Price: 80
+                      
+                    }
+                  },
+                  {
+                    id: 3,
+                    details: {
+                      Quantity:0,
+                      Name: "Samosa",
+                      Image:"https://images.unsplash.com/photo-1601050690597-df0568f70950?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8M3x8c2Ftb3NhfGVufDB8fDB8fA%3D%3D&w=1000&q=80",
+                      Description:
+                        "Allo Stuffed Dish",
+                      Price: 40
+                     
+                    },
+                  },
+            ]
+        }
+    }
+      
+  buttonClicked(detail){
+    let newCart=this.state.cart;
+    if(detail.Quantity === 0){
+    detail.Quantity=detail.Quantity+1;
+    newCart.push(detail);
+    }
+    else
```

Comment:
const newCartValue = getNewCartValue(detail)
this.setState({cart: newCartValue})

In helper file
function getNewCartValue() { 
   // update the cart without mutation, you can use the map function here. : if item available in cart
}

You can also use copy. Use spread operator to copy.